src/app/user/registration/registration.component.ts

```diff
@@ -9,31 +10,20 @@ import { ToastrComponentlessModule, ToastrService } from 'ngx-toastr';
 })
 export class RegistrationComponent implements OnInit {
 
-  constructor(public service : UserService,private toastr : ToastrService) { }
+  constructor(public service : UserService,private toastr : ToastrService, public user : UserComponent) { }
 
   ngOnInit(): void {
   }
 
   onSubmit() {
     this.service.register().subscribe(
```

Comment:
Technically, your registration object should be created in this file. Once it is created, send it to the service function as a parameter.

Reply:
done